test/Microsoft.IdentityModel.SampleTests/SampleTokenValidationClassTests.cs

```diff
@@ -221,6 +222,218 @@ public void TokenWithMissingSecurityCredentials()
         }
         #endregion
 
+        #region New Model Token Validation Tests
+        /// <summary>
+        /// Tests how the class under test handles a valid token.
+        /// </summary>
+        [Fact]
+        public void ValidToken_NewPath()
+        {
+            SampleTokenValidationClass classUnderTest = new SampleTokenValidationClass();
+            classUnderTest.ValidateTokenShimWithNewPath(testTokenCreator.CreateDefaultValidToken());
+        }
+
+        /// <summary>
+        /// Tests how the class under test handles a bogus token; one that in no way conforms to the expected JWS format.
+        /// </summary>
+        [Fact]
+        public void BogusToken_NewPath()
+        {
+            TestWithGeneratedToken_NewPath(
+                () => "InvalidToken",
+                typeof(SecurityTokenMalformedException),
+                "IDX14100");
+        }
+
+        /// <summary>
+        /// Tests how the class under test handles a token with a missing signature.
+        /// </summary>
+        [Fact]
+        public void TokenWithoutSignature_NewPath()
+        {
+            TestWithGeneratedToken_NewPath(
+                testTokenCreator.CreateTokenWithNoSignature,
+                typeof(SecurityTokenInvalidSignatureException),
+                "IDX10504:");
+        }
+
+        /// <summary>
+        /// Tests how the class under test handles a token with a malformed signature.
+        /// </summary>
+        [Fact]
+        public void TokenWithBadSignature_NewPath()
+        {
+            TestWithGeneratedToken_NewPath(
+                testTokenCreator.CreateTokenWithInvalidSignature,
+                typeof(SecurityTokenInvalidSignatureException),
+                "IDX10500:"); // 10500 indicates no signature key was found. Current path returns 10511 which indicates a bad signature and provides the list of keys attempted
+        }
+
+        /// <summary>
+        /// Tests how the class under test handles a token which is expired.
+        /// </summary>
+        [Fact]
+        public void ExpiredToken_NewPath()
+        {
+            TestWithGeneratedToken_NewPath(
+                testTokenCreator.CreateExpiredToken,
+                typeof(SecurityTokenExpiredException),
+                "IDX10223");
+        }
+
+        /// <summary>
+        /// Tests how the class under test handles a token which is not yet valid
+        /// </summary>
+        [Fact]
+        public void NetYetValidToken_NewPath()
```

Comment:
This is too fractured.
We need sets of tests using TheoryData for Lifetime, Audience, etc.
Why not part of JsonWebTokenHandler.ValidateLifetimeTests